bin/test/main.ml

```diff
@@ -353,25 +353,33 @@ let run_exn (`Setup ()) (`Non_deterministic non_deterministic)
       | None ->
         run_toplevel_tests ?root c ppf tests t
   in
-
-  Mdx.run ?syntax ~force_output file ~f:(fun file_contents items ->
-      let temp_file = Filename.temp_file "ocaml-mdx" ".output" in
-      at_exit (fun () -> Sys.remove temp_file);
-      let buf = Buffer.create (String.length file_contents + 1024) in
-      let ppf = Format.formatter_of_buffer buf in
-      List.iter (function
-          | Section _
-          | Text _ as t -> Mdx.pp_line ?syntax ppf t
-          | Block t ->
-            List.iter (fun (k, v) -> Unix.putenv k v) (Block.set_variables t);
-            try
-              Mdx_top.in_env (Block.environment t)
-                (fun () -> test_block ~ppf ~temp_file t)
-            with Failure msg ->
-              raise (Test_block_failure (t, msg))
-        ) items;
-      Format.pp_print_flush ppf ();
-      Buffer.contents buf);
+  let _ = Misc.remove_file in
+  let gen_expected file_contents items =
+    let temp_file = Filename.temp_file "ocaml-mdx" ".output" in
+    at_exit (fun () -> Sys.remove temp_file);
+    let buf = Buffer.create (String.length file_contents + 1024) in
+    let ppf = Format.formatter_of_buffer buf in
+    List.iter (function
+        | Section _
+        | Text _ as t -> Mdx.pp_line ?syntax ppf t
+        | Block t ->
+          List.iter (fun (k, v) -> Unix.putenv k v) (Block.set_variables t);
+          try
+            Mdx_top.in_env (Block.environment t)
+              (fun () -> test_block ~ppf ~temp_file t)
+          with Failure msg ->
+            raise (Test_block_failure (t, msg))
+      ) items;
+    Format.pp_print_flush ppf ();
+    Buffer.contents buf
+  in
+  let run_to_file outfile =
+    Mdx.run_to_file ?syntax ~force_output ~outfile ~f:gen_expected file
+  in
+  (match (output : Cli.output option) with
+   | Some Stdout -> Mdx.run_to_stdout ?syntax ~f:gen_expected file
+   | Some (File outfile) -> run_to_file outfile
+   | None -> run_to_file (file ^ ".corrected"));
```

Comment:
You should call `Mdx.run` here. It does the `^ ".corrected"`.

Reply:
`Mdx.run` is still in the API for compatibility reasons, I'd rather not use it internally and use the more up-to-date equivalent instead.

Ideally we'll deprecate it in a little while!